### PR TITLE
feat(reader-data): check newsletter subscription on login

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

If a contact is already signed up for a list in the ESP, then somehow registers for the site with the same email address, they are not segmented as a newsletter subscriber until they update their subscriptions (by either signing up for a new list or unsubscribing from one in My Account).

TL;DR the segmentation update isn't triggered until you edit subscriptions on the site

Closes `1200550061930446/1205326345667208`.

### How to test the changes in this Pull Request:

1. Check out `master`.
2. Either turn off RAS temporarily on your site and sign up for a newsletter, or manually add a contact to a list directly in the connected ESP.
3. On the Newspack site, register for an account using the same email address. without signing up for newsletters. This can be done via a Registration block if multiple lists are available and you deselect them all when registering, or by making a donation.
4. Navigate around the site and observe you are not segmented as a newsletter subscriber, despite being subscribed in the ESP.
5. Go to My Account and verify if needed. Observe you can see the newsletter list signup status despite still not being segmented properly.
6. Update your subscriptions by signing up for a new list, or unsubscribing from a list and then resubscribing.
7. Continue navigating around the site and observe you're now segmented properly. 
8. Check out this branch. Repeat steps 2-3, but this time confirm that you're segmented properly as a newsletter subscriber immediately after registering for or logging into an account.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/reader-activation/class-reader-data.php [hunk 1]
```diff
@@ -330,5 +332,34 @@ public static function setup_reader_activity() {
...
+		}
+		// Bail if reader is already a newsletter subscriber.
+		if ( $is_newsletter_subscriber ) {
+			return;
+		}
+		// Check if the user is subscribed to a newsletter.
+		if ( ! class_exists( '\Newspack_Newsletters' ) || ! class_exists( '\Newspack_Newsletters_Subscription' ) ) {
+			return;
+		}
+		$subscribed_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $email_address );
```
Comment (dkoo) [inline includes/reader-activation/class-reader-data.php:359]:
```suggestion
		$subscribed_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $data['email'] );
```

**Review by dkoo — CHANGES_REQUESTED** [review]
This is working with the suggested change.

**Comment by miguelpeixe** [comment]
Nice!

**Review by dkoo — APPROVED** [review]
